### DISCUSSION TITLE
[Bug]: Cannot update or upload assets after deploying pimcore to PROD stand

### DISCUSSION THREAD

**Discussion by mikbagautdinov** [pimcore/pimcore#13296] [Q&A / Support]
### Expected behavior

The Operator is able to upload, update and work with assets.

### Actual behavior

When trying to upload new file the message is shown:

_An exception occurred while executing 'SELECT IFNULL(binaryFileId, id) FROM versions WHERE binaryFileHash = ? AND cid = ? AND storageType = ? ORDER BY id ASC LIMIT 1' with params ["d5b38a8ce2e5bb509ac1b5864ef5d8aeff2f7f8dee876a2688e6d61dd14732ee409ddfacd33eb3d35b00f709a89efb4625a3a49416d62660672421c3ee38dc57", 415, "fs"]: SQLSTATE[42S22]: Column not found: 1054 Unknown column 'storageType' in 'where clause'_

When trying to update files - "Error" message without any specification is shown 

### Steps to reproduce

We have a Stage stand working properly. We are deploying it on a PROD stand based on Pimcore Skeleton configuration using the following Steps:

1. Dump deployment to the database docker container
2. Files Copying to the php-fpm docker container at paths: 
- /var/www/html/public/var/assets,
- /var/www/html/var/classes,
- /var/www/html/var/config/datahub-configurations.php
![error1](https://user-images.githubusercontent.com/104482192/193515631-6111e3d4-0006-4ac0-831a-f2d393c10d6b.png)

When I try to get any of the assets by http request I get an error
![error1](https://user-images.githubusercontent.com/104482192/193515711-4f744cea-c26b-4b26-b1ea-4948d0772346.png)
When I try to upload a new version of the file I also get an error
![upload_new_version](https://user-images.githubusercontent.com/104482192/193515869-09b88064-7e2b-48ce-9da0-23f4d7f302dc.png)
When I try to upload a new file an error occurs
![upload_new_file](https://user-images.githubusercontent.com/104482192/193515968-e7210d5b-ace0-43c3-b7f3-06f9708619df.png)
Though I can freely download any of the assets and this operation is always successfull.

**Comment by mcop1** [comment]
@mikbagautdinov did you check if there is a column called `storageType` in the `versions` table? The column should be created during installation or migration, please see: https://github.com/pimcore/pimcore/pull/11458

**Comment by mikbagautdinov** [comment]
@mcop1 There is no column storageType in the versions table either on the stage stand (Pimcore Platform v10.1.3) or on PROD stand (Pimcore Platform v10.5.5) after database dump was deployed. I assume it is because of the versions difference. As there was no such problem with assets which I could add,upload version and so on before database dump was deployed. 
How can I get the type of data I must use when adding a new column storageType to the versions table? 
I will be grateful for help!

**Comment by mcop1** [comment]
It is ok if the column isn´t there in v10.1.3, it was introduced in 10.4.0 with a corresponding migration:
https://github.com/pimcore/pimcore/blob/b2110bb3c416cbf4a7fce967fe98a0e031d642a0/bundles/CoreBundle/src/Migrations/Version20220214110000.php#L36-L37

**Comment by mikbagautdinov** [comment]
Ok, thank you @mcop1 
I got it fixed partly by adding column storageType to the versions table. The data type of the column and its length was found at the path \vendor\pimcore\pimcore\bundles\InstallBundle\Resources\install.sql
Now I have one problem -I can't get assets except images (jpg, png). 
![getAsset4](https://user-images.githubusercontent.com/104482192/193565549-50be3acb-bc88-4a3f-bf81-8fe5ce893d4f.png)
![getAsset](https://user-images.githubusercontent.com/104482192/193562259-f2cf9804-bb05-40a6-a9d3-a3e1955ee192.png)
![getAsset2](https://user-images.githubusercontent.com/104482192/193562628-1e841be5-07cb-46e8-865e-58b625dbad0d.png)
![getAsset3](https://user-images.githubusercontent.com/104482192/193562652-dd09eb45-416d-4798-b62a-2629edad12ec.png)

**Reply by mikbagautdinov** [reply]
@mcop1  Is there any way to fix the above? The problem occurs when I work with assets except pdf, jpg,png and probably other "standard" files. 
I have mentioned that I am trying to settle everything on skeleton configuration, e.g. dumped database and some of the catalogs copied to the php container.
I also see a question mark near the file name in assets.
Thank you for your time
![image](https://user-images.githubusercontent.com/104482192/193799748-dbe123dc-67b1-44d0-b674-1ee64c8cd86a.png)

**Reply by mcop1** [reply]
@mikbagautdinov Hello,
the "?" icon is displayed because pimcore doesn´t recognize the file type (gltf). The other error, I assume, has something to do with the setup of your productive environment, I am afraid I can´t help you there.
